### ISSUE TITLE
Building from sources failed on Mac

### ISSUE THREAD

**Issue by ShogunWeb** [Audiveris/audiveris#673]
Hi,

When building Audiveris on my Mac, I have this error : 

-----------------------------------------------
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/jeremiefays/audiveris/settings.gradle'

* What went wrong:
Could not compile settings file '/Users/jeremiefays/audiveris/settings.gradle'.
> startup failed:
  General error during conversion: Unsupported class file major version 64
-----------------------------------------------

I have openjdk version "20.0.2" installed on my Mac, and Gradle 8.3

Could it be due to having a later version than java 17 installed on my computer ? Also, I see that even though Gradle 8.3 is installed on my computer, the build starts by installing gradle 7.3 (which is perhaps incompatible with java 20)

Any idea on hom to solve this problelm ?

**Comment by hbitteur** [comment]
**Gradle 7.3** comes from the file `./gradle/wrapper/gradle-wrapper.properties` in your project directory.
But I don't think this is the problem.

You should try to use **Java 17** instead of Java 20.

**Comment by CaTwoPlus** [comment]
Using Java 17 works when getting "Unsupported class file major version 65" as well.

**Comment by hbitteur** [comment]
FYI:
- on the "development" branch, we today use Gradle 8.5 and Java 21.
- this does not apply yet to the "master" branch (and to the installed version of course, because it is always based on the master branch). On this branch we are still with Gradle 7.3 and Java 17.

**Comment by hbitteur** [comment]
@jamesellinger 

1/ Regarding the problem with OCR, the very beginning of your log file says:
> 2024-03-19 21:11:59,226 INFO  []              TesseractOCR 116  | TESSDATA_PREFIX value: null
2024-03-19 21:11:59,226 WARN  []              TesseractOCR 344  | Tesseract data could not be found. Try setting TESSDATA_PREFIX environment variable to point to tessdata folder.
2024-03-19 21:11:59,226 INFO  []              TesseractOCR 241  | OCR folder: null

In plain English, the Tesseract library could not find the needed language data files.
Please have a look at [this handbook section](https://audiveris.github.io/audiveris/_pages/install/languages/) which explains how to download these language data files to some location of yours, and then set TESSDATA_PREFIX environment variable so that it points to your chosen location.
You are not the first to be bitten by this! A real fix is still on my own TODO list, see issue #692.

2/ Regarding the problem with Mac environment, the log file gives:
> 2024-03-19 21:11:59,815 WARN  []            MacApplication 220  | Unable to setup Mac OS X GUI integration
java.lang.IllegalAccessException: class org.audiveris.omr.ui.MacApplication cannot access class com.apple.eawt.Application (in module java.desktop) because module java.desktop does not export com.apple.eawt to unnamed module @673ba40c

I personally don't have access to a Mac, so I can't help you directly here. But the error message looks familiar. This appeared with Java 9, when the Java implementation was reorganized, with a stronger "protection" against access to the internal modules. Of course, not all software around the world could instantly adapt to the new rules.
To get around this, you must sign a kind of “waiver.”

See these lines in the file `build.gradle` located at the root of Audiveris project, around line 71:
>    // Needed to circumvent the limitations brought by Java 9 and above (the JigSaw cuts)
    applicationDefaultJvmArgs = ["--add-exports=java.desktop/sun.awt.image=ALL-UNNAMED"]

I suppose you have to use something similar, perhaps:
>   applicationDefaultJvmArgs = ["--add-exports=java.desktop/com.apple.eawt=ALL-UNNAMED"]

But, like I said, I cannot check by myself.
So, please give it a try and let us know about the results.

**Comment by jamesellinger** [comment]
Thank you!

Regarding OCR, that's on me since I should have read the handbook more closely.

Reagrding the second issue, changing line 71 in build.gradle to 

`applicationDefaultJvmArgs = ["--add-exports=java.desktop/com.apple.eawt=ALL-UNNAMED"]`

did not change the error message. I saw that you posted the same suggestion in [#564](https://github.com/Audiveris/audiveris/issues/564#issuecomment-1537193602) a few months ago. Whatever the source is, it's still unresolved (I also just compiled Audiveris on an older Intel-based apple and get the same error). 

Anyway, as I said it doesn't seem to have an impact on function. I can open PDFs and export to .mxl. I appreciate your work creating and maintaing this software.

<img width="1544" alt="Screenshot 2024-03-21 at 15 30 52" src="https://github.com/Audiveris/audiveris/assets/28816526/7cbeeb0d-d50a-4f5a-b719-9d5fbc50657d">

**Comment by domq** [comment]
So, what is the timeline to switch the `master` branch to a recent version of Gradle?

In the meantime, I think that the `README.md` should clarify that `master` won't build on Mac OS X, unless one downgrades Java somehow.

```bash
$ sw_vers

ProductName:		macOS
ProductVersion:		15.1.1
BuildVersion:		24B91

$ /usr/bin/javac -version

javac 21.0.4
```
